docs/orchestrating-elastic-stack-applications/maps.asciidoc

```diff
@@ -0,0 +1,245 @@
+
+The following sections describe how to customize an {ems} deployment to suit your requirements.
+
+* <<{p}-maps-es,Connect to an Elasticsearch cluster>>
+* <<{p}-maps-data>>
+** <<{p}-maps-basemap-download,Basemap Download>>
+** <<{p}-maps-pod-configuration,Pod Configuration>>
+* <<{p}-maps-advanced-configuration,Advanced configuration>>
+** <<{p}-maps-configuration>>
+** <<{p}-maps-scaling>>
+* <<{p}-maps-http-configuration,HTTP Configuration>>
+** <<{p}-maps-http-publish,Load balancer settings and TLS SANs>>
+** <<{p}-maps-http-custom-tls,Provide your own certificate>>
+** <<{p}-maps-http-disable-tls,Disable TLS>>
+** <<{p}-maps-ingress>>
+
+[id="{p}-maps-es"]
+== Connect to an Elasticsearch cluster
+
+{ems} needs a connection to Elasticseach to verify the installed license. You define the connection via the `elasticsearchRef` attribute:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: quickstart
+    namespace: default
+----
+
+The use of `namespace` is optional if the Elasticsearch cluster is running in the same namespace as {ems}.
+
+NOTE: Any {ems} can reference (and thus access) any Elasticsearch instance as long as they are both in namespaces that are watched by the same ECK instance. ECK will copy the required Secret from Elasticsearch to the {ems} namespace. {ems} cannot automatically connect to Elasticsearch (through `elasticsearchRef`) in a namespace managed by a different ECK instance. For more information, see <<{p}-restrict-cross-namespace-associations,Restrict cross-namespace resource associations>>.
+
+The {ems} configuration file is automatically setup by ECK to establish a secure connection to Elasticsearch.
+[id="{p}-maps-data"]
+== Map data
+The {ems} Docker image contains only a few zoom levels of data. In order to get the map data up to the highest zoom level {ems} needs a basemap file mounted into its container.
+[id="{p}-maps-basemap-download"]
+=== Basemap download
+You have to download the basemap ahead of time on a machine that is not air-gapped and populate a volume that can be mounted into the {ems} Pods.   See also the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server[{ems} documentation.]
+
+The exact procedure of how to get a Kubernetes volume populated with that data is out of scope of this documenation as it depends on your specific Kubernetes setup and choice of volume provisioner. One possible approach that should work for most setups is as follows:
+
+. Download the basemap zip archive using the link  shown in the {ems} UI or extracted from the  `/status` endpoint.
+. Create a PersistentVolumeClaim of sufficient size (> 90G) and a temporary Pod to mount the corresponding volume
```

Comment:
I guess we don't want to load the data in the EMS Pod because of the `readOnly: true` section. I am wondering how much mandatory the `readOnly` setting is. Do we expect the EMS containers to concurrently write in the volume ? (I'm just wondering if there is a way to avoid this temporary Pod)

Reply:
I think it is not mandatory at all. Let me see if we can simplify this further.

Reply:
The problem is that we mount on top of the basemap file that comes with the Docker image. So initially there will be no basemap file until we manage to copy it into the volume. However the Pod will boot loop because the container cannot run without a basemap.

Reply:
👍 , makes sense, thank you for the clarification.